java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/BuildPipelineConfigurator.java

```diff
@@ -102,6 +102,7 @@
     StageDockerArtifactBuildStep stageDockerBuildStep
         = buildStepFactory.createStageDockerArtifactBuildStep();
     stageDockerBuildStep.setArtifactPathOverride(runtimeConfig.getArtifact());
+    stageDockerBuildStep.setRuntimeConfig(runtimeConfig);
```

Comment:
I think it might make more sense for `runtimeConfig` to be injected into the constructor of the build steps that need it. I tend to like that better, since there's more of a guarantee that the dependency will be there when needed. 

Since the yaml parsing happens after the initial injection time, you may need to use the `buildStepFactory` to provide a `RuntimeConfig` instance to those classes using the `@Assisted` annotation:

https://github.com/google/guice/wiki/AssistedInject#assistedinject-in-guice-30